packages/react-core/src/components/ChipGroup/ChipGroupToolbarItem.tsx

```diff
@@ -19,47 +20,93 @@ export interface ChipGroupToolbarItemProps extends React.HTMLProps<HTMLUListElem
+  tooltipPosition?: 'auto' | 'top' | 'bottom' | 'left' | 'right';
 }
 
-export const ChipGroupToolbarItem: React.FunctionComponent<ChipGroupToolbarItemProps> = ({
-  categoryName = '',
-  children = null,
-  className = '',
-  isClosable = false,
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  onClick = (_e: React.MouseEvent) => undefined as any,
-  closeBtnAriaLabel = 'Close chip group',
-  ...props
-}: ChipGroupToolbarItemProps) => {
-  if (React.Children.count(children)) {
-    const renderChipGroup = (id: string, HeadingLevel: any) => (
-      <ul className={css(styles.chipGroup, styles.modifiers.toolbar, className)} {...props}>
-        <li>
-          <HeadingLevel className={css(styles.chipGroupLabel)} id={id}>
-            {categoryName}
-          </HeadingLevel>
-          <ul className={css(styles.chipGroup)}>{children}</ul>
-          {isClosable && (
-            <div className="pf-c-chip-group__close">
-              <ChipButton
-                aria-label={closeBtnAriaLabel}
-                onClick={onClick}
-                id={`remove_group_${id}`}
-                aria-labelledby={`remove_group_${id} ${id}`}
-              >
-                <TimesIcon aria-hidden="true" />
-              </ChipButton>
-            </div>
-          )}
-        </li>
-      </ul>
-    );
+interface ChipGroupToolbarItemState {
+  isTooltipVisible: boolean;
+}
+
+export class ChipGroupToolbarItem extends React.Component<ChipGroupToolbarItemProps, ChipGroupToolbarItemState> {
+  constructor(props: ChipGroupToolbarItemProps) {
+    super(props);
+    this.state = {
+      isTooltipVisible: false
+    };
+  }
+
+  heading = React.createRef<HTMLHeadingElement>();
+
+  static defaultProps: ChipGroupToolbarItemProps = {
+    categoryName: '',
+    children: null,
+    className: '',
+    isClosable: false,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    onClick: (_e: React.MouseEvent) => undefined as any,
+    closeBtnAriaLabel: 'Close chip group',
+    tooltipPosition: 'top'
+  };
 
-    return (
-      <ChipGroupContext.Consumer>
-        {(HeadingLevel: any) => <GenerateId>{randomId => renderChipGroup(randomId, HeadingLevel)}</GenerateId>}
-      </ChipGroupContext.Consumer>
-    );
+  componentDidMount() {
+    this.setState({
+      isTooltipVisible: Boolean(
+        this.heading.current && this.heading.current.offsetWidth < this.heading.current.scrollWidth
+      )
+    });
   }
-  return null;
-};
+
+  render() {
+    const {
+      categoryName,
+      children,
+      className,
+      isClosable,
+      closeBtnAriaLabel,
+      onClick,
+      tooltipPosition,
+      ...rest
+    } = this.props;
+
+    if (React.Children.count(children)) {
+      const renderChipGroup = (id: string, HeadingLevel: any) => (
+        <ul className={css(styles.chipGroup, styles.modifiers.toolbar, className)} {...rest}>
+          <li>
+            {this.state.isTooltipVisible ? (
```

Comment:
feature changes should be in a separate PR

Reply:
This change comes from merging `master` into `v4`.

Reply:
This change comes from merging `master` into `v4`.